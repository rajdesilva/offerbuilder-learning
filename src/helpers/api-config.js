import { init as initApm } from "@elastic/apm-rum";

export const baseURL = {
  API_URL_BASE: process.env.REACT_APP_BASE_API_URL,
  LOGIN_URL: process.env.REACT_APP_LOGIN_URL,
  LOGOUT_URL: process.env.REACT_APP_LOGOUT_URL,
  PROFILE_URL: process.env.REACT_APP_USER_PROFILE_URL,
  HISTORY_URL: process.env.REACT_APP_USER_HISTORY_URL,
};

export const apm = initApm({
  // Set required service name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
  serviceName: process.env.REACT_APP_APM_SERVICE_NAME,
  serverUrl: process.env.REACT_APP_APM_SERVICE_URL,

  // Set service version (required for sourcemap feature)
  serviceVersion: process.env.REACT_APP_APM_SERVICE_VERSION,
});