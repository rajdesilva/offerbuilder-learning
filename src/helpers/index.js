export * from './api-config';
export * from './history';
