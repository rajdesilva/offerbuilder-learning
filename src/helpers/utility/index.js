export { createSearchQueryURL } from './createSearchQueryURL';
export { isParentOrChildrenPropertyAdded } from './isParentOrChildrenPropertyAdded';
export { isPropertySelected } from './isPropertySelected';
export { mapBrands } from './mapBrands';
export { mapOnlySuppliers } from './mapOnlySuppliers';
export { parseFormDescriptionsAndImagesForOffer } from './parseFormDescriptionsAndImagesForOffer';
export { removePropertyForSelectedChannels } from './removePropertyForSelectedChannels';
export { removePropertyForSelectedSuppliers } from './removePropertyForSelectedSuppliers';
export { getParsedProperty } from './getParsedProperty';
export { getFormattedValue } from './getFormattedValue';
export { getValue } from './getValue';
export { parsePropertyMarketingInfoToFormData } from './parsePropertyMarketingInfoToFormData';
export { createOfferRequestObject } from './createOfferRequestObject';
export { getParsedOffers } from './getParsedOffers';
export { getListOfPropertiesForSupplier } from './getListOfPropertiesForSupplier';
export { getListOfPropertiesForChannels } from './getListOfPropertiesForChannels';
export { getInitializedState } from './getInitializedState';
export { getParsedNewOfferSettingsObject } from './getParsedNewOfferSettingsObject';
export { getListOfPropertiesForStorefront } from './getListOfPropertiesForStorefront';
export { checkIfMultipleChannelPresentForTarget } from './checkIfMultipleChannelPresentForTarget';
export { checkIfMultipleSupplierPresentForTarget } from './checkIfMultipleSupplierPresentForTarget';
export { checkIfMultipleStorefrontPresentForTarget } from './checkIfMultipleStorefrontPresentForTarget';
export { removePropertyForSelectedStorefront } from './removePropertyForSelectedStorefront';
export { checkIfMultipleBrandPresentForTarget } from './checkIfMultipleBrandPresentForTarget';
export { getListOfPropertiesForBrand } from './getListOfPropertiesForBrand';
export { getSearchStateFromRedux } from './getSearchStateFromRedux';
export { getObjectStringify } from './getObjectStringify';
export { parseOfferMarketingFromOffer } from './parseOfferMarketingFromOffer';
export { parseOfferMarketingLanguagesFromOffer } from './parseOfferMarketingLanguagesFromOffer';
export { getFilterData } from './getFilterData';
export { getSearchAndFilterQueryURL } from './getSearchAndFilterQueryURL';
export { getSearchQueryURLForEditAddProperties } from "./getSearchQueryURLForEditAddProperties";
export { isSavedOfferProperty } from './isSavedOfferProperty';
export { getParsedPropertiesForDeeplinkSettings } from "./getParsedPropertiesForDeeplinkSettings";
export { resetApplicationReduxAndStorage } from "./resetApplicationReduxAndStorage";
export { updateSettingsBrandWithSearchedValues } from "./updateSettingsBrandWithSearchedValues";
export { checkIfUserHasRole } from "./checkIfUserHasRole";
export { getUserOfferBuilderRole } from "./getUserOfferBuilderRole";
export { checkIfNewPropertiesAdded } from "./checkIfNewPropertiesAdded";
export { getAllBrandsWithoutKeysFromTempBrands } from "./getAllBrandsWithoutKeysFromTempBrands";
export { checkIfItemAlreadyPresentInList } from "./checkIfItemAlreadyPresentInList";
export { checkIfTravellingDateValidAsPerBookingDate } from "./checkIfTravellingDateValidAsPerBookingDate";
export { checkIfPropertiesHaveMultipleLocations } from "./checkIfPropertiesHaveMultipleLocations";
export { displayMargin } from "./displayMargin";
export { getFilterCount } from "./getFilterCount";
export { labelForBrands } from "./labelForBrands";
export { getDistanceFromLatLonInKm } from "./getDistanceFromLatLonInKm";
export { getTrustYouIdListFromProperties } from "./getTrustYouIdListFromProperties";
export { getSearchParamsToFetchMarketPrice } from "./getSearchParamsToFetchMarketPrice";
export { getMarketPriceForPropertyFromList } from "./getMarketPriceForPropertyFromList";
export { getImageFilterCount } from "./getImageFilterCount";
export { getSearchAndFilterImagesQueryUrl } from "./getSearchAndFilterImagesQueryUrl";
export { getImageName } from "./getImageName";
export { updateReduxWithOfferData } from "./updateReduxWithOfferData";
export { getOfferTypeFromPropertiesSelected } from "./getOfferTypeFromPropertiesSelected";