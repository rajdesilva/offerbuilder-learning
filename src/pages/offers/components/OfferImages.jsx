import React, { Fragment, useEffect, useState } from "react";
import { Row, Col, List, Button, Image } from "antd";
import { SortableContainer, SortableElement } from "react-sortable-hoc";
import arrayMove from "array-move";
import { FormattedMessage } from "react-intl";
import defaultImg from "../../../assets/images/default-img.svg";
import { CloseOutlined } from "@ant-design/icons";
import OfferImageSelectionModal from "./OfferImageSelectionModal";
import UploadImage from "./UploadImage";
import OfferImagesCarousel from "./OfferImagesCarousel";
import { useDispatch, useSelector } from "react-redux";
import isEqual from "lodash.isequal";
import { cloneDeep } from "lodash";
import styles from "./css/OfferImages.module.less";
import { offerImageSearchAndFilterActions } from "../actions/offerImageSearchAndFilterActions";

const SortableItem = SortableElement(({ value }) => (
  <List.Item key={value.index} className={styles["marketing-image-hover"]}>
    <Image
      src={value.image.url}
      className="list-image"
      onError={(ev) => (ev.target.src = defaultImg)}
    ></Image>
    <Button
      data-testid={value.index + "-remove-image-btn"}
      ghost
      type="text"
      className="remove-image-btn"
      icon={<CloseOutlined className="remove-img-icon" />}
      onClick={() => value.removeImageFromIndex(value.index)}
    ></Button>
  </List.Item>
));

const DisplayImageList = SortableContainer(({ value }) => {
  return (
    <Image.PreviewGroup>
      <List
        grid={{ gutter: 16, column: 10 }}
        dataSource={value.uploadedImages}
        className={styles["list-item-img"]}
        renderItem={(image, index) => (
          <SortableItem
            value={{
              image: image,
              index: index,
              removeImageFromIndex: value.removeImageFromIndex,
            }}
            key={`drag-item-${index}`}
            index={index}
          />
        )}
      />
    </Image.PreviewGroup>
  );
});

export default function OfferImages(props) {
  const dispatch = useDispatch();
  const reduxSelectedImages = useSelector(
    (state) =>
      window.getValue(state, "newoffermarketinginfo.marketingInfo.images") ||
      [],
    isEqual
  );
  const offerImageList = useSelector(
    (state) =>
      window.getValue(state, "offerimagesearchandfilters.offerImageList") || [],
    isEqual
  );

  const [carouselImage, setCarouselImage] = useState({
    index: 0,
    display: false,
    isSelected: false,
  });
  const [uploadedImages, setUploadedImages] = useState(
    reduxSelectedImages || []
  );
  const [uploadDetails, setUploadDetails] = useState({
    displayImageSelectorModal: false,
    displayUpload: false,
  });
  const [selectedImages, setSelectedImages] = useState(reduxSelectedImages);

  useEffect(() => {
    setUploadedImages(reduxSelectedImages);
  }, [reduxSelectedImages]);

  const hideModal = () => {
    setUploadDetails({
      displayImageSelectorModal: false,
      displayUpload: false,
    });
    dispatch({
      type: offerImageSearchAndFilterActions.APPLY_OFFER_IMAGE_INITIAL_STATE,
    });
  };

  const refreshSelectedImages = () => {
    setSelectedImages(reduxSelectedImages);
  };

  const updateSelectedImages = (imageSelected, isToAdd) => {
    if (isToAdd) {
      const temp = cloneDeep(selectedImages);
      temp.push({ ...imageSelected });

      setSelectedImages(temp);
    } else {
      const temp = selectedImages.filter(
        (image) => image.id !== imageSelected.id
      );

      setSelectedImages(temp);
    }
  };

  const setUpdatedSelectedImages = (updatedImages) => {
    setSelectedImages(updatedImages);
  };

  const removeImageFromIndex = (removedIndex) => {
    const updatedArray = uploadedImages.filter(
      (word, index) => index !== removedIndex
    );
    setUploadedImages(updatedArray);
    props.onChange(updatedArray);
  };

  const onSortEnd = ({ oldIndex, newIndex }) => {
    const updatedArray = arrayMove(uploadedImages, oldIndex, newIndex);
    props.onChange(updatedArray);
    setUploadedImages(updatedArray);
  };

  const toggleUploadModal = (updatedStatus, uploadedFilesArray) => {
    if (window.getValue(uploadedFilesArray, "length") > 0) {
      setSelectedImages(selectedImages.concat(uploadedFilesArray));
    }
    setUploadDetails({
      displayImageSelectorModal: !updatedStatus,
      displayUpload: updatedStatus,
    });
  };

  const displaySelectedImages = () => {
    props.onChange(selectedImages);
    setUploadedImages(selectedImages);
  };

  const setCarouselDetails = (index, display, isSelected) => {
    setCarouselImage({
      index,
      display,
      isSelected,
    });
  };

  const hideCarousel = () => {
    setCarouselImage({
      index: 0,
      isSelected: false,
      display: false,
      images: [],
    });
  };

  return (
    <Fragment>
      {uploadDetails.displayImageSelectorModal ? (
        <OfferImageSelectionModal
          hideModal={hideModal}
          toggleUploadModal={toggleUploadModal}
          displaySelectedImages={displaySelectedImages}
          setCarouselImage={setCarouselDetails}
          updateSelectedImages={updateSelectedImages}
          setUpdatedSelectedImages={setUpdatedSelectedImages}
          selectedImages={selectedImages}
          refreshSelectedImages={refreshSelectedImages}
        />
      ) : (
        <Fragment />
      )}
      {carouselImage.display ? (
        <OfferImagesCarousel
          carouselImage={carouselImage}
          selectedImages={selectedImages}
          displaySelectedImages={displaySelectedImages}
          updateSelectedImages={updateSelectedImages}
          images={carouselImage.isSelected ? selectedImages : offerImageList}
          hideCarousel={hideCarousel}
        />
      ) : (
        <Fragment />
      )}
      {uploadDetails.displayUpload ? (
        <UploadImage
          toggleUploadModal={toggleUploadModal}
        />
      ) : (
        <Fragment />
      )}

      <Row gutter={[8, 8]}>
        <Col span={4}>
          <FormattedMessage id="nemo.images" />
        </Col>
        <Col span={20}>
          {uploadedImages && uploadedImages.length === 0 ? (
            <Fragment />
          ) : (
            <DisplayImageList
              value={{
                uploadedImages: uploadedImages,
                removeImageFromIndex: removeImageFromIndex,
              }}
              onSortEnd={onSortEnd}
              axis="x"
              distance={1}
            />
          )}
          <Button
            ghost
            type="primary"
            data-testid="manage-offer-image-btn"
            onClick={() => {
              setUploadDetails({
                displayImageSelectorModal: true,
                displayUpload: uploadDetails.displayUpload,
              });
            }}
          >
            <FormattedMessage id="nemo.manageImages" />
          </Button>
        </Col>
      </Row>
    </Fragment>
  );
}
