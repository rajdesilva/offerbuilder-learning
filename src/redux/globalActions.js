export const globalActions = {
  'SET_SUCCESS': 'SET_SUCCESS',
  'SET_ERROR': 'SET_ERROR',
  'SET_LANGUAGE': 'SET_LANGUAGE',
  'CLEAR': 'CLEAR',
}