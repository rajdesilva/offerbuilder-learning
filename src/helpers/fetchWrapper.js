import { message } from "antd";
import { createIntl, createIntlCache } from "@formatjs/intl";
import * as languages from "../langs/";
import { baseURL } from "./api-config";
import { history } from "./history";

// This is optional but highly recommended
// since it prevents memory leak
const cache = createIntlCache();

// Create the `intl` object
const intl = createIntl(
  {
    // Locale of the application
    // TODO:
    locale: "en",
    // Locale of the fallback defaultMessage
    defaultLocale: "en",
    messages: languages["en"],
  },
  cache
);
const defaultRequest = {
  credentials: "include",
  headers: {
    "Content-Type": "application/json",
  },
};
export const fetchWrapper = (url, payload) => {
  payload = {
    ...defaultRequest,
    ...payload,
  };
  return fetch(url, payload).then(
    (response) => {
      // Check that the response is valid and reject an error
      // response to prevent subsequent attempt to parse json
      if (!response.ok) {
        if (response.status === 500) {
          message.error(
            intl.formatMessage({
              id: "nemo.error500msg",
            })
          );
        } else if (response.status === 503) {
          message.error(
            intl.formatMessage({
              id: "nemo.error503msg",
            })
          );
        } else if (response.status === 504) {
          message.error(
            intl.formatMessage({
              id: "nemo.error504msg",
            })
          );
        } else {
          message.error(
            intl.formatMessage({
              id: "nemo.somethingWentWrongPleaseTryAgain",
            })
          );
        }
        return;
      }
      if (response.status === 403) {
        history.push("/access-denied");
      }
      return response.json ? response.json() : response;
    },
    (error) => {
      /**
       * user url, payload and error here for sending to the logger.
       */
      if (error instanceof TypeError) {
        window.location.href = baseURL.API_URL_BASE + "redirect";
      }
      console.error("fetch error : ", error);
    }
  );
};
