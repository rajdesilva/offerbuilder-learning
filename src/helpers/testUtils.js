import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { getValue } from "./utility";
import rootReducers from "../redux/rootReducers";
import * as languages from "../langs";
import { IntlProvider } from "react-intl";
const initialReducerState = {};
window.getValue = getValue;

window.google = {
  maps: {
    Marker: class {},
    Map: class {},
    LatLngBounds: class {},
    places: {
      Autocomplete: class {},
      AutocompleteService: class {},
      PlacesServiceStatus: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
      PlacesAutocomplete: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },

    MarkerClusterer: class {},
    Geocoder: class {},
  },
};
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

function render(
  ui,
  {
    initialState = initialReducerState,
    store = createStore(rootReducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <IntlProvider locale="en" messages={languages["en"]}>
          {children}
        </IntlProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
