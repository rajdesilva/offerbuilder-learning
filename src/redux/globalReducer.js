import { globalActions } from "./globalActions";
import produce from "immer";

const initialState = {
  error: "",
  success: "",
  language: "",
};
export const globalReducer = produce((draft, action) => {
  switch (action.type) {
    case globalActions.SET_ERROR:
      draft.success = "";
      draft.error = action.payload;
      return;
    case globalActions.SET_SUCCESS:
      draft.error = "";
      draft.success = action.payload;
      return;
    case globalActions.SET_LANGUAGE:
      draft.language = localStorage.getItem('language')
      return;
    case globalActions.CLEAR:
      return initialState;
    default:
      return draft;
  }
}, initialState);
