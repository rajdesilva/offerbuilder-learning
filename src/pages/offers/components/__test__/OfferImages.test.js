import React from "react";
import { createStore } from "redux";
import { render, fireEvent } from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import OfferImages from "../OfferImages";

import { tempStore } from "./tempStore";
import { cloneDeep } from "lodash";
import { act } from "react-test-renderer";

window.fetchWrapper = window.fetchWrapper = jest.fn().mockResolvedValue({
  success: true,
});
jest.mock("./../../service");

describe("OfferImages component test", () => {
  test("create OfferImages components snapshot", () => {
    const selectImageStore = cloneDeep(tempStore);
    selectImageStore.newoffermarketinginfo.marketingInfo.images = [
      {
        id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
        uploadedBy: "swapnil.deshmukh@kognitiv.com",
        checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
        url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
        createdAt: "2021-03-30T09:45:15.200651",
        updatedAt: "2021-03-30T09:45:15.200651",
        offerDetails: [],
      },
    ];
    const store = createStore(() => ({ ...selectImageStore }));
    store.dispatch = jest.fn();
    const props = {
      onChange: jest.fn(),
      images: [
        {
          id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
          createdAt: "2021-03-30T09:45:15.200651",
          updatedAt: "2021-03-30T09:45:15.200651",
          offerDetails: [],
        },
      ],
    };
    const { container } = render(
      <MemoryRouter>
        <OfferImages {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    expect(container).toMatchSnapshot();
  });

  test("OfferImages components- click on manage images btn and popup is displayed", async () => {
    const selectImageStore = cloneDeep(tempStore);
    selectImageStore.newoffermarketinginfo.marketingInfo.images = [
      {
        id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
        uploadedBy: "swapnil.deshmukh@kognitiv.com",
        checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
        url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
        createdAt: "2021-03-30T09:45:15.200651",
        updatedAt: "2021-03-30T09:45:15.200651",
        offerDetails: [],
      },
    ];
    const store = createStore(() => ({ ...selectImageStore }));
    store.dispatch = jest.fn();
    const props = {
      onChange: jest.fn(),
      images: [
        {
          id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
          createdAt: "2021-03-30T09:45:15.200651",
          updatedAt: "2021-03-30T09:45:15.200651",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 2-46-07 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "8db052ac1d98a3a5a28eb9c1a53ecb8f",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688146/offer_builder_dev/Screenshot%202021-03-24%20at%202-46-07%20PM.png",
          createdAt: "2021-03-25T17:02:26.832568",
          updatedAt: "2021-03-25T17:02:26.832568",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 1-30-49 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "7634f719f11b8d5673b85b3707029863",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688107/offer_builder_dev/Screenshot%202021-03-24%20at%201-30-49%20PM.png",
          createdAt: "2021-03-25T17:01:47.516244",
          updatedAt: "2021-03-25T17:01:47.516244",
          offerDetails: [],
        },
      ],
    };
    const { getByTestId, getByText } = render(
      <MemoryRouter>
        <OfferImages {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const manageBtn = getByTestId("manage-offer-image-btn");
    expect(manageBtn).toBeTruthy();
    await act(async () => {
      await fireEvent.click(manageBtn);

      expect(getByText(/Selecting images for/i)).toBeInTheDocument();
    });
  });

  test("OfferImages components- click on remove image btn and onchange is called", async () => {
    const selectImageStore = cloneDeep(tempStore);
    selectImageStore.newoffermarketinginfo.marketingInfo.images = [
      {
        id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
        uploadedBy: "swapnil.deshmukh@kognitiv.com",
        checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
        url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
        createdAt: "2021-03-30T09:45:15.200651",
        updatedAt: "2021-03-30T09:45:15.200651",
        offerDetails: [],
      },
    ];
    const store = createStore(() => ({ ...selectImageStore }));
    store.dispatch = jest.fn();
    const props = {
      onChange: jest.fn(),
      images: [
        {
          id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
          createdAt: "2021-03-30T09:45:15.200651",
          updatedAt: "2021-03-30T09:45:15.200651",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 2-46-07 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "8db052ac1d98a3a5a28eb9c1a53ecb8f",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688146/offer_builder_dev/Screenshot%202021-03-24%20at%202-46-07%20PM.png",
          createdAt: "2021-03-25T17:02:26.832568",
          updatedAt: "2021-03-25T17:02:26.832568",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 1-30-49 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "7634f719f11b8d5673b85b3707029863",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688107/offer_builder_dev/Screenshot%202021-03-24%20at%201-30-49%20PM.png",
          createdAt: "2021-03-25T17:01:47.516244",
          updatedAt: "2021-03-25T17:01:47.516244",
          offerDetails: [],
        },
      ],
    };
    const { getByTestId } = render(
      <MemoryRouter>
        <OfferImages {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const removeImgBtn = getByTestId("0-remove-image-btn");
    expect(removeImgBtn).toBeTruthy();
    await act(async () => {
      await fireEvent.click(removeImgBtn);
      expect(props.onChange).toHaveBeenCalledTimes(1);
    });
  });

  test("OfferImages components- remove btn with only one image displayed as per redux/edit values", async () => {
    const selectImageStore = cloneDeep(tempStore);
    selectImageStore.newoffermarketinginfo.marketingInfo.images = [
      {
        id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
        uploadedBy: "swapnil.deshmukh@kognitiv.com",
        checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
        url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
        createdAt: "2021-03-30T09:45:15.200651",
        updatedAt: "2021-03-30T09:45:15.200651",
        offerDetails: [],
      },
    ];
    const store = createStore(() => ({ ...selectImageStore }));
    store.dispatch = jest.fn();
    const props = {
      onChange: jest.fn(),
      images: [
        {
          id: "Screenshot 2021-03-30 at 1-14-53 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "60eef42bf8d3b2a6578f59001af2fcf7",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png",
          createdAt: "2021-03-30T09:45:15.200651",
          updatedAt: "2021-03-30T09:45:15.200651",
          offerDetails: [],
        },
        {
          id: "Screenshot 2021-03-24 at 2-46-07 PM.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "8db052ac1d98a3a5a28eb9c1a53ecb8f",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688146/offer_builder_dev/Screenshot%202021-03-24%20at%202-46-07%20PM.png",
          createdAt: "2021-03-25T17:02:26.832568",
          updatedAt: "2021-03-25T17:02:26.832568",
          offerDetails: [],
        },
      ],
    };
    const { getByTestId, queryByTestId } = render(
      <MemoryRouter>
        <OfferImages {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const removeImgBtn = getByTestId("0-remove-image-btn");
    expect(removeImgBtn).toBeTruthy();
    expect(getByTestId("0-remove-image-btn")).toBeInTheDocument();
    expect(queryByTestId("1-remove-image-btn")).not.toBeInTheDocument();
  });
});
