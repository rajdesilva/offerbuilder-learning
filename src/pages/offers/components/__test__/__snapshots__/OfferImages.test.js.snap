// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferImages component test create OfferImages components snapshot 1`] = `
<div>
  <div
    class="ant-row"
    style="margin: -4px -4px 4px -4px;"
  >
    <div
      class="ant-col ant-col-4"
      style="padding: 4px 4px 4px 4px;"
    >
      Images
    </div>
    <div
      class="ant-col ant-col-20"
      style="padding: 4px 4px 4px 4px;"
    >
      <div
        class="ant-list ant-list-split ant-list-grid"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-row"
              style="margin-left: -8px; margin-right: -8px;"
            >
              <div
                style="width: 10%; max-width: 10%;"
              >
                <div
                  class="ant-col"
                  style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
                >
                  <div
                    class="ant-list-item"
                  >
                    <div
                      class="ant-image"
                    >
                      <img
                        class="ant-image-img list-image"
                        src="https://res.cloudinary.com/seekda-dev/image/upload/v1617090314/offer_builder_dev/Screenshot%202021-03-30%20at%201-14-53%20PM.png"
                      />
                    </div>
                    <button
                      class="ant-btn ant-btn-text ant-btn-icon-only remove-image-btn"
                      data-testid="0-remove-image-btn"
                      type="button"
                    >
                      <span
                        aria-label="close"
                        class="anticon anticon-close remove-img-icon"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="close"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-primary ant-btn-background-ghost"
        data-testid="manage-offer-image-btn"
        type="button"
      >
        Add Images
      </button>
    </div>
  </div>
</div>
`;
